Added coverage to --retry

Mendi03 opened the pull request:
Did the following to add coverage for --retry:

- Added a test_retry method under the [tests/cli/test_build.py](url) file
- Modified the logic of the run under the [buildtest/cli/build.py](url), [buildtest/builders/base.py](url) and [buildtest/executors/local.py](url) files

I did not rewrite lines 563-583 under [buildtest/cli/build.py](url) since it caused errors with test_build_limit.

This is the code I tried using:

```
# if (timeout) or (limit is not None) or (retry is not None):
#   list1 = [timeout, limit, retry]
#   for x in list1:
#       if not isinstance(x, int):
#           raise BuildTestError(f"{x} is not of type int")
#       if x <= 0:
#           raise BuildTestError(f"{x} must be greater than 0")
```

Mendi03 committed:
Added coverage for retry and modified run method logic

Changed multiple files so that method test_retry would run properly.

Mendi03 committed:
Formatted code using style -a

Small commit to ignore .idea file from pycharm and reformatted some code to fit the style

Mendi03 committed:
Update .gitignore

Mendi03 committed:
removed .idea

removed .idea and added it to the .gitignore

Mendi03 committed:
Removed commented code

Removed the following comment from the build.py file:

# if (timeout) or (limit is not None) or (retry is not None):
        #   list1 = [timeout, limit, retry]
        #   for x in list1:
        #       if not isinstance(x, int):
        #           raise BuildTestError(f"{x} is not of type int")
        #       if x <= 0:
        #           raise BuildTestError(f"{x} must be greater than 0")

shahzebsiddiqui commented on .gitignore:
```diff
@@ -46,4 +46,7 @@ env.bak/
+coverage.json
+
+# ignore .idea
+*.idea
```
i think it should just be `.idea` instead of `*.idea` PyCharm creates a top-level directory named `.idea` so this should exclude all files.

If you want to put a comment then you can say `ignore IntelliJ’s project settings files`

shahzebsiddiqui replied:
okay please remove this since @szuananwar has added this in https://github.com/buildtesters/buildtest/pull/1497

shahzebsiddiqui commented on buildtest/cli/build.py:
```diff
@@ -566,16 +566,21 @@ def __init__(
+            if not isinstance(retry, int):
+                raise BuildTestError(f"{retry} is not of type int")
+            if retry <= 0:
+                raise BuildTestError("Retry must be greater than 0")
```
so do a for loop something like and use the same if condition and just replace the entry with the `field`. Semantically it is the same code just more efficient.

```python
     for field in [timeout, limit, retry]:
        if field is not None:
           ...

Mendi03 committed:
Rewrote code in _init__ method for better semantics

Wrote a for loop under the _init__ method in buildtest class for better semantics

Mendi03 committed:
Fixed broken url for stale

Changed https://github.com/marketplace/stale

to https://github.com/marketplace/actions/close-stale-issues in the file docs/contributing/github_integration.rst

shahzebsiddiqui commented on buildtest/cli/build.py:
```diff
@@ -563,18 +563,12 @@ def __init__(
+                if not isinstance(field, int):
+                    raise BuildTestError(f"{field} is not of type int")
+                if field <= 0:
+                    raise BuildTestError("Limit must be greater than 0")
```
change `Limit` -> `{field}`

Mendi03 replied:
Thanks for catching that! Should be fixed now.

Mendi03 committed:
Fixed error in error message

shahzebsiddiqui reviewed (approved):
thanks @Mendi03 this looks good and congrats on your 1st contribution!
